CMakeLists.txt

```diff
@@ -24,7 +24,7 @@ if(NOT WT_ARCH)
     # Defer to our hosts architecture as our target architecture.
     # PROCESSOR_ARCHITECTURE env variable could be unset on certain Windows systems,
     # in that case map the architecture to x86.
-    if ("${CMAKE_HOST_SYSTEM_PROCESSOR}" MATCHES "^(x86_64|i686|i386|AMD64|)$")
+    if ("${CMAKE_HOST_SYSTEM_PROCESSOR}" MATCHES "^(x86_64|i686|AMD64|)$")
```

Comment:
`i686` is a 32bit architecture which [WT doesn't support](https://github.com/wiredtiger/wiredtiger/blob/ea46937c577c6c004896edae958b69ab43806fc7/src/include/gcc.h#L55).
It might warrant further discussion/investigation before doing so, but I'd like to drop it.

Reply:
I am in favour of dropping it, @korteland thoughts?

Reply:
I think this can be dropped. Andrew's correct about it being 32-bit and thus unsupported, and it also spuriously appeared during the Autotools -> CMake migration.

Reply:
Dropped: 42b45966b